doc: update AUTHORS list

targos opened the pull request:
Update AUTHORS list using tools/update-authors.sh.

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, cluster, crypto, etc). -->

targos commented:
There may be a few duplicates but this adds 282 new collaborators since the last update that was 4 months ago 🎉

targos commented on AUTHORS:
```diff
@@ -624,7 +624,6 @@ Ray Donnelly <mingw.android@gmail.com>
 dead-horse <dead_horse@qq.com>
 Luis Reis <luis.m.reis@gmail.com>
 sudodoki <smd.deluzion@gmail.com>
-Steven Loomis <srl@icu-project.org>
```
duplicate of line 614

targos commented on AUTHORS:
```diff
@@ -971,7 +970,6 @@ James Lal <james@silklabs.com>
 Josh Leder <josh@ha.cr>
 Surya Panikkal <surya.com@gmail.com>
 vsemozhetbyt <vsemozhetbyt@gmail.com>
-Eugene Obrezkov <ghaiklor@gmail.com>
```
duplicate of line 955

targos commented on AUTHORS:
```diff
@@ -1247,7 +1246,6 @@ Ethan Arrowood <ethan.arrowood@gmail.com>
 Dan Villa <danielavilla02@gmail.com>
 CodeTheInternet <edlerner+github@protonmail.com>
 Eric Gonzalez <ericxgonzalez@gmail.com>
-sxa555 <sxa@uk.ibm.com>
```
duplicate of line 788

targos commented on AUTHORS:
```diff
@@ -1338,7 +1335,6 @@ John Barboza <jbarboza@ca.ibm.com>
 Paul Graham <homosaur@gmail.com>
 Nate <nfoxley@stedwards.edu>
 Chris Story <storytimesolutions@gmail.com>
-Stewart X Addison <sxa@uk.ibm.com>
```
ditto

lpinca commented on AUTHORS:
```diff
@@ -1415,5 +1411,287 @@ Bradley Curran <brad@bradleycurran.com>
+liusi <siyuan.liu@autodesk.com>
+Noj Vek <noj.vek@mixpanel.com>
+Ruslan Bekenev <furyinbox@gmail.com>
+Danny Nemer <hi@DannyNemer.com>
```
dup.

lpinca commented on AUTHORS:
```diff
@@ -1415,5 +1411,287 @@ Bradley Curran <brad@bradleycurran.com>
+Chris Burkhart <christide@christide.com>
+Rj Bernaldo <rj.bernaldo@gmail.com>
+John F. Mercer <john.f.mercer@gmail.com>
+Pedro lima <pvsousalima@gmail.com>
```
dup.

lpinca commented on AUTHORS:
```diff
@@ -1415,5 +1411,287 @@ Bradley Curran <brad@bradleycurran.com>
+Griffith Tchenpan <griffith.tchenpan@myworkpapers.com>
+Oky Antoro <kiantrue@gmail.com>
+icarter09 <icarter1391@gmail.com>
+Saad Quadri <saad@saadq.com>
```
dup.

TimothyGu commented on AUTHORS:
```diff
@@ -1415,5 +1410,284 @@ Bradley Curran <brad@bradleycurran.com>
+Artur Vieira <vieira.artur.g@gmail.com>
+Flarna <deb2001-github@yahoo.de>
+kysnm <tokyoincidents.g@gmail.com>
+Artur G Vieira <vieira.artur.g@gmail.com>
```
Dup of 1504.

TimothyGu commented on AUTHORS:
```diff
@@ -1415,5 +1410,284 @@ Bradley Curran <brad@bradleycurran.com>
+Madara Uchiha <me@madara.ninja>
+Gil Tayar <gil@tayar.org>
+Glenn Schlereth <Glenn.Schlereth@gmail.com>
+Michael Dawson <mdawson@devrus.com>
```
*Maybe* a dupe? @mhdawson can you confirm?

targos replied:
It's a dupe

TimothyGu commented on AUTHORS:
```diff
@@ -1415,5 +1410,284 @@ Bradley Curran <brad@bradleycurran.com>
+Artur G Vieira <vieira.artur.g@gmail.com>
+Sreepurna Jasti <jsreepur@in.ibm.com>
+Rafael Fragoso <rafaelfragosom@gmail.com>
+daniel.bevenius <daniel.bevenius@gmail.com>
```
Dupe.

TimothyGu commented on AUTHORS:
```diff
@@ -1415,5 +1410,284 @@ Bradley Curran <brad@bradleycurran.com>
+Ankit Parashar <ankitabc12@gmail.com>
+James Kyle <me@thejameskyle.com>
+Daniil Shakir <phpinfo12345@gmail.com>
+davidmarkclements <huperekchuno@googlemail.com>
```
Dupe.

TimothyGu commented on AUTHORS:
```diff
@@ -1415,5 +1410,280 @@ Bradley Curran <brad@bradleycurran.com>
+Christian d'Heureuse <chdh@inventec.ch>
+Shahar Or <mightyiampresence@gmail.com>
+detailyang <detailyang@gmail.com>
+jBarz <jbarboza@ca.ibm.com>
```
Dupe.

TimothyGu commented on AUTHORS:
```diff
@@ -1415,5 +1410,280 @@ Bradley Curran <brad@bradleycurran.com>
+Uppinder Chugh <uppinderchugh@gmail.com>
+Karl Cheng <qantas94heavy@gmail.com>
+Taylor Woll <taylor.woll@microsoft.com>
+taylor.woll <tawoll@ntdev.microsoft.com>
```
Dupe of the line above.

TimothyGu commented on AUTHORS:
```diff
@@ -1415,5 +1410,280 @@ Bradley Curran <brad@bradleycurran.com>
+Paul Bininda <paul@bininda.com>
+sreepurnajasti <sreepurna.jasti@gmail.com>
+Gautam Mittal <200mittalgautam@gmail.com>
+Jamen Marz <jamenmarz+gh@gmail.com>
```
Dupe of 1543

TimothyGu commented on AUTHORS:
```diff
@@ -1415,5 +1410,280 @@ Bradley Curran <brad@bradleycurran.com>
+Mandeep Singh <mandeep.singh@zomato.com>
+Prakash Palaniappan <prakaash.kpk@gmail.com>
+Keita Akutsu <kakts.git@gmail.com>
+Gareth Ellis <gareth@gsellis.com>
```
Dupe. Different email address though.

benjamingr commented on AUTHORS:
```diff
@@ -1415,5 +1410,276 @@ Bradley Curran <brad@bradleycurran.com>
+Steven Lehn <sjlehn@gmail.com>
+Łukasz Szewczak <lukaszewczak@gmail.com>
+Madara Uchiha <me@madara.ninja>
+Gil Tayar <gil@tayar.org>
```
@giltayar @MadaraUchiha

benjamingr commented on AUTHORS:
```diff
@@ -1415,5 +1410,276 @@ Bradley Curran <brad@bradleycurran.com>
+Rob Wu <rob@robwu.nl>
+Steven Winston <swinston100@hotmail.com>
+sallen450 <jqh101@sina.com>
+OriLev <sviatos@gmail.com>
```
@OriLev

benjamingr commented on AUTHORS:
```diff
@@ -1415,5 +1410,276 @@ Bradley Curran <brad@bradleycurran.com>
+Zhang Weijie <email@zhangweijie.cn>
+Gunar C. Gessner <gunar@gunargessner.com>
+SkyAo <csvwolf@qq.com>
+Benjamin Gruenbaum <benji@peer5.com>
```
This is most likely a duplicate of inglor@gmail.com (work mail vs. old personal mail)

silverwind commented:
Instead of removing duplicates, you should use `.mailmap` to alias them instead. That way, the next time this file is regenerated, you don't have to dedupe the same users again.

LakshmiSwethaG commented:
@targos, thanks for this. I have since then changed my organization, and therefore the email IDs too. Can you please replace `lgopired@in.ibm.com` with `lakshmiswethagopireddy@gmail.com`?
 thanks in advance."

targos commented:
@silverwind do you know how it works? Because for example Ben keeps being added even though all his addresses are in the `.mailmap`

silverwind commented:
@targos I'm pretty sure that Ben's entry in `.mailmap`is invalid:

```
Ben Noordhuis <info@bnoordhuis.nl> <bnoordhuis@bender.(none)> <ben@strongloop.com>
```

Acorrding to the [shortlog docs](https://www.kernel.org/pub/software/scm/git/docs/git-shortlog.html#_mapping_authors), a mailmap entry can either contain a mail adress or a mail adress and a name, so corrected, it should be:

````
Ben Noordhuis <info@bnoordhuis.nl> <bnoordhuis@bender.(none)>
Ben Noordhuis <info@bnoordhuis.nl> <ben@strongloop.com>
````

From a quick look, I think this is the only error in the current `.mailmap`.

targos commented:
@silverwind Thank you, I understand now. Updating...

bidipyne commented:
@targos, thanks a lot. Since then, I have  changed my organization, and therefore email IDs too. Can you please replace my mail IDs `bidipyne@in.ibm.com` to `pyne.bidisha2017@gmail.com`? 
Thank You!

200GAUTAM commented:
@targos, thanks a lot. Since then, I have changed my organization, and therefore email IDs too. Can you please replace my mail IDs gaumit34@in.ibm.com to 200mittalgautam@gmail.com?
Thank You!

200GAUTAM commented:
@targos, thanks a lot. Since then, I have changed my organization, and therefore email IDs too. Can you please replace my mail IDs bidipyne@in.ibm.com to pyne.bidisha2017@gmail.com?
Thank You!

200GAUTAM commented:
@targos, thanks a lot. Since then, I have changed my organization, and therefore email IDs too. Can you please replace my mail IDs gaumit34@in.ibm.com to 200mittalgautam@gmail.com?
Thank You!

sreepurnajasti commented on AUTHORS:
```diff
@@ -1415,5 +1410,276 @@ Bradley Curran <brad@bradleycurran.com>
+Artur Vieira <vieira.artur.g@gmail.com>
+Flarna <deb2001-github@yahoo.de>
+kysnm <tokyoincidents.g@gmail.com>
+Sreepurna Jasti <jsreepur@in.ibm.com>
```
@targos As I have changed my organization, I would request you to update this with my personal email id: **sreepurna.jasti@gmail.com**. Thanks in advance.

sreepurnajasti commented on AUTHORS:
```diff
@@ -1415,5 +1410,276 @@ Bradley Curran <brad@bradleycurran.com>
+Alexander O'Mara <me@alexomara.com>
+James, please <jamesplease2@gmail.com>
+Josh Ferge <josh@augur.io>
+sreepurnajasti <jsreepur@in.ibm.com>
```
@targos As I have changed my organization, I would request you to update this with my personal email id: **sreepurna.jasti@gmail.com**.

Thanks in advance.

targos commented:
@LakshmiSwethaG, @bidipyne, @sreepurnajasti  I changed your emails in the map.
@200GAUTAM I think it's already fine for you?

targos commented:
@lpinca @TimothyGu @XadillaX @mhdawson @silverwind I made a significant change since last review. Now the mailmap is up to date and running `tools/update-authors.sh` creates the AUTHORS file as it is in my branch.

TimothyGu commented on .mailmap:
```diff
@@ -128,35 +157,48 @@ Kelsey Breseman <ifoundthemeaningoflife@gmail.com>
 Onne Gorter <onne@onnlucky.com>
 Paul Querna <pquerna@apache.org> <paul@querna.org>
+Pedro Lima <pvsousalima@gmail.com> Pedro Victor <pvsousalima@gmail.com>
+Pedro Lima <pvsousalima@gmail.com> Pedro Lima <pvsousalima@gmail.com>
```
Is this line needed?

targos replied:
It is, but I made a typo. It's a version of the name with lowercase "L". Weird that it was working with that mistake...

200GAUTAM commented:
@targos yes 200mittalgautam@gmail.com was already there , Sorry for the confusion

targos commented:
I searched for all remaining duplicate email addresses in AUTHORS and added them to the map

TimothyGu commented on AUTHORS:
```diff
@@ -1102,16 +1102,14 @@ Michael-Rainabba Richardson <rainabba@gmail.com>
 Rene Weber <weber.rene@live.de>
 Lauren Spiegel <lhspiegel@gmail.com>
-Lydia Katsamberis <llkats@gmail.com>
+Lydia Kats <llkats@gmail.com>
```
I assume the first one is a more formal name?

targos replied:
I chose the one that is on their GitHub profile: https://github.com/llkats

targos commented:
Landed in https://github.com/nodejs/node/commit/3f7813b0e9d9ea5c98d1e8a130d2b63119320924
